src/Symfony/Bundle/TwigBundle/CacheWarmer/TemplateCacheWarmer.php

```diff
@@ -34,26 +36,48 @@ class TemplateCacheWarmer implements CacheWarmerInterface, ServiceSubscriberInte
     public function __construct(
         private ContainerInterface $container,
         private iterable $iterator,
+        private ?CacheInterface $cache = null,
     ) {
     }
 
     public function warmUp(string $cacheDir, ?string $buildDir = null): array
     {
         $this->twig ??= $this->container->get('twig');
 
-        foreach ($this->iterator as $template) {
-            try {
-                $this->twig->load($template);
-            } catch (Error) {
+        $originalCache = $this->twig->getCache();
+        if ($originalCache instanceof NullCache) {
```

Comment:
Also when the cache is disabled with `$cache === false`.
```suggestion
        if (!originalCache || $originalCache instanceof NullCache) {
```

Reply:
Is it necessary though ? The entire TemplateCacheWarmer is unregistered when the cache is disabled.

Reply:
Indeed, I missed that point.